### PR TITLE
Hema

### PR DESCRIPTION
**Pull request by hemavmane**
Graph

### CODE REVIEW

File: src/Component/CovidGraph/CovidGraph.jsx [hunk 1]
```diff
@@ -8,54 +9,66 @@ import {
...
-              "vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com",
-          },
-        }
-      )
+            'X-RapidAPI-Key': '8c0c5645f4msh17b4e89de7b874ap10eb20jsn3e95f4dad049',
+            'X-RapidAPI-Host': 'vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com'
+          }
+        })
       .then(response => {
         console.log(response.data);
```
Comment (Souvikdeb2612) [inline src/Component/CovidGraph/CovidGraph.jsx]: remove console statements before raising pr

File: src/Component/CovidGraph/CovidGraph.jsx [hunk 2]
```diff
@@ -8,54 +9,66 @@ import {
...
+
+        
       })
       .catch(function (error) {
         console.error(error);
-      });
+      
+    }
+      );
+      console.log(coviddata)
```
Comment (Souvikdeb2612) [inline src/Component/CovidGraph/CovidGraph.jsx]: Same as above

File: src/Component/CovidGraph/CovidGraph.jsx [hunk 3]
```diff
@@ -8,54 +9,66 @@ import {
...
 export function MyCovidApp() {
+  
   const [coviddata, setCovidData] = useState([]);
-
+// let obj =[{
+//   country:"india",
+//   TotalCases:"983493",
+//   TotalDeaths:"32490"
+// }]
   useEffect(() => {
```
Comment (Souvikdeb2612) [inline src/Component/CovidGraph/CovidGraph.jsx]: Don't call the data here directly. Call the data in the other component where you import the chart and pass data as props.

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/Component/CovidGraph/CovidGraph.jsx (hunk 3 above)
Comment (hemavmane) [reply src/Component/CovidGraph/CovidGraph.jsx]: okay sir

File: src/Component/ContextApi/ContextApi.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          },
+        }
+      )
+      .then(response => {
+        console.log(response.data);
+        response.data = response.data.sort();
+        setCovidData([...response.data]);
+      })
+      .catch(function (error) {
+        console.error(error);
```
Comment (Souvikdeb2612) [inline src/Component/ContextApi/ContextApi.js]: remove console statement before pushing code
Handle errors with toaster or simple messages in DOM

File: src/Component/ContextApi/ContextApi.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          headers: {
+            "X-RapidAPI-Key":
+              "8c0c5645f4msh17b4e89de7b874ap10eb20jsn3e95f4dad049",
+            "X-RapidAPI-Host":
+              "vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com",
+          },
+        }
+      )
+      .then(response => {
+        console.log(response.data);
```
Comment (Souvikdeb2612) [inline src/Component/ContextApi/ContextApi.js]: remove console statement

File: src/Component/CovidGraph/CovidGraph.jsx [hunk 6]
```diff
@@ -9,65 +7,23 @@ import {
   Tooltip,
   Area,
 } from "recharts";
+import { useContext } from "react";
+import {createContextApi}  from "../ContextApi/ContextApi"
```
Comment (Souvikdeb2612) [inline src/Component/CovidGraph/CovidGraph.jsx]: Give 1 line space after imports

File: src/Component/CovidGraph/CovidGraph.jsx [hunk 7]
```diff
@@ -9,65 +7,23 @@ import {
...
-      );
-      console.log(coviddata)
-  }, []);
-
-  
   return (
     <>
       <h1>COVID CASES & DEATHS</h1>
-      <div style={{margin:"100px"}}>
+      <div style={{margin:"130px",padding:"10px",overflow:"visible"}}>
```
Comment (Souvikdeb2612) [inline src/Component/CovidGraph/CovidGraph.jsx]: remove inline styling

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/Component/ContextApi/ContextApi.js (hunk 4 above)
Comment (hemavmane) [reply src/Component/ContextApi/ContextApi.js]: noted

File: src/Component/ContextApi/ContextApi.js (hunk 5 above)
Comment (hemavmane) [reply src/Component/ContextApi/ContextApi.js]: resolved

File: src/Component/CovidGraph/CovidGraph.jsx (hunk 6 above)
Comment (hemavmane) [reply src/Component/CovidGraph/CovidGraph.jsx]: resolved

File: src/Component/CovidGraph/CovidGraph.jsx (hunk 7 above)
Comment (hemavmane) [reply src/Component/CovidGraph/CovidGraph.jsx]: resolved

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**